src/modules/common/PlayerFrame.tsx

```diff
@@ -19,84 +20,94 @@ const PlayerFrame = (): JSX.Element => {
     name: SettingsKeys.SettingsData,
   });
 
-  const [labels, setLabels] = useState<DraggableLabelType[]>([]);
+  const [answeredLabels, setAnsweredLabels] = useState<AnsweredLabel[]>([]);
+  // labels will be null only before setting the state as we cannot render all labels within container if not settled yet
+  const [labels, setLabels] = useState<null | Label[]>(null);
 
   const [isDragging, setIsDragging] = useState(false);
 
   useEffect(() => {
-    const appLabels = appSettings?.[0].data.labels;
-    const imageDimension = appSettings?.[0].data.imageDimension;
-
-    if (imageDimension) {
-      if (appLabels) {
-        const labelsP = appLabels.map((l, index) => ({
-          labelId: l.id,
-          ind: index + 1,
-          choices: [],
-          x: `${(l.x / imageDimension.width) * 100}%`,
-          y: `${(l.y / imageDimension.height) * 100}%`,
-        }));
-
-        const allChoices = appLabels.map(({ id, content }) => ({
-          id,
-          content,
-        }));
-
-        const allLabels = [
-          {
-            ind: 0,
-            y: '0%',
-            x: '0%',
-            labelId: 'all-labels',
-            choices: allChoices,
-          },
-          ...labelsP,
-        ];
-        setLabels(allLabels);
-      }
+    const settingLabels = appSettings?.[0].data.labels;
+
+    if (settingLabels) {
+      const answered = settingLabels.map((label) => ({
+        expected: label,
+        actual: null,
+      }));
+
+      setAnsweredLabels(answered);
+      setLabels(settingLabels);
     }
   }, [appSettings]);
 
+  const findAnLabelIndex = (id: string): number =>
+    answeredLabels?.findIndex(({ expected }) => expected.id === id) ?? -1;
+
   const onDragEnd = (draggable: DropResult): void => {
```

Comment:
`onDragEnd` is a bit long, what do you think about extracting some of its logic in smaller functions ?